pandas/core/internals/managers.py

```diff
@@ -1220,36 +1220,42 @@ def value_getitem(placement):
             if inplace and blk.should_store(value):
                 # Updating inplace -> check if we need to do Copy-on-Write
                 if using_copy_on_write() and not self._has_no_reference_block(blkno_l):
-                    blk.set_inplace(blk_locs, value_getitem(val_locs), copy=True)
-                    self._clear_reference_block(blkno_l)
+                    nbs_tup = tuple(blk.delete(blk_locs))
+                    first_nb = new_block_2d(
+                        value_getitem(val_locs), BlockPlacement(blk.mgr_locs[blk_locs])
+                    )
+                    if self.refs is not None:
+                        self.refs.extend([self.refs[blkno_l]] * len(nbs_tup))
```

Comment:
Do we need to clear the refs at ``self.refs[blkno_l]`` now, since first_nb should not have a reference block?

Reply:
Thx for catching, must have removed the line accidentally

Reply:
Adjusted one of the tests to cover this